handler/incoming_instance_handler.go

```diff
@@ -109,6 +111,24 @@ func (hdlr *InstanceEventHandler) validate(newInstance event.NewInstance) error
 }
 
 func (hdlr *InstanceEventHandler) insertDimensions(ctx context.Context, instance *model.Instance, dimensions []*model.Dimension) error {
+	// validate instance
```

Comment:
Hi David
On Monday, please can you walk me thru the new code as to how the mongo actions are separated from all graph actions ?

Reply:
yes, no problem! I separated the tasks in 2 funcs, one that is executed for each option by a go-routine, and one that is executed afterwards (once per batch) which gets the order values from graph db and does the patch to mongoDB. We can have a call.